### PR TITLE
Organize prototyper and container tool prompts

### PR DESCRIPTION
**Pull request by DonggeLiu**
Restructure and decouple prompts, add clearer instructions and steps.

Although we should keep this, there are still problems that cannot be fixed by simple prompting.
E.g., LLM cannot figure out which `*.o` files to add to include the function-under-test, even though there are examples from other fuzz targets.

I will do this via more agents later.

### CODE REVIEW

**Comment by DonggeLiu** [comment]
Report looking good:
https://llm-exp.oss-fuzz.com/Result-reports/ofg-pr/2024-10-28-675-dg-comparison/index.html

**Review by oliverchang — APPROVED** [review]

File: agent/prototyper.py [hunk 1]
```diff
@@ -163,13 +174,31 @@ def _container_handle_conclusion(
...
       logger.info('***** Failed to recompile in %02d rounds *****', cur_round)
       prompt_text = (
-          'Failed to build fuzz target. Here is the fuzz target, build'
-          ' script, compliation command, and other compilation runtime'
-          ' output.\n<fuzz target>\n'
-          f'{build_result.fuzz_target_source}\n</fuzz target>\n'
-          f'<build script>\n{build_result.build_script_source}\n'
-          f'</build script>\n<compilation log>\n{compile_log}\n'
-          '</compilation log>\n')
+          'Failed to build fuzz target. Here is the fuzz target, build script, '
```
[the remark this answers is not in the document]
Comment (DonggeLiu) [reply agent/prototyper.py:177]: Sure!

[the remark this answers is not in the document]
Comment (DonggeLiu) [reply agent/prototyper.py:177]: Let me solve this in the next PR, which I will further break down prototyper into more single-tasked agents.
Those agents will likely have their individual prompts, I will write them in individual files too.

**Pull request merged**